src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala

```diff
@@ -24,156 +27,159 @@ case class CromwellRuntimeAttributes(docker: Option[String],
-
-    validatedRuntimeAttributes match {
-      case Success(r) => r
-      case Failure(f) =>
-        throw new IllegalArgumentException() with ThrowableWithErrors {
-          val message = "RuntimeAttribute is not valid."
-          val errors = f
-        }
+      *  Warn if keys are found that are unsupported on this backend.  This is not necessarily an error, at this point
+      *  the keys are known to be supported on at least one backend other than this.
+      */
+    unsupportedKeys(wdlRuntimeAttributes.attrs.keys, backendType) foreach log.warn
+    val requiredKeysNel = validateRequiredKeys(wdlRuntimeAttributes.attrs, backendType)
+    val unknownKeysNel = validateUnknownKeys(wdlRuntimeAttributes.attrs, backendType)
+    (requiredKeysNel |@| unknownKeysNel) { (_, _) } match {
+      case scalaz.Success(x) => scala.util.Success(wdlRuntimeAttributes)
+      case scalaz.Failure(e) => scala.util.Failure(new IllegalArgumentException(e.list.mkString("\n")))
     }
   }
 
+  def unsupportedKeys(keys: Iterable[String], backendType: BackendType): Seq[String] = {
+    val supportedKeys = backendType.supportedKeys map { _.key }
+    val unsupportedKeys = keys.toSet -- supportedKeys
+
+    if (unsupportedKeys.isEmpty) Vector.empty
+    else Vector(s"Found unsupported keys for backend '$backendType': " + unsupportedKeys.toSeq.sorted.mkString(", "))
+  }
+
   val LocalDiskName = "local-disk"
-  val LocalizationDisk = LocalDisk(LocalDiskName, DiskType.SSD, Option(10)).toDisk
-
-  /** Fallback values if values for these keys are not specified in a task "runtime" stanza. */
-  object Defaults {
-    val Cpu = 1L
-    val Disk = Vector(LocalizationDisk)
-    val FailOnStderr = false
-    val ContinueOnReturnCode = ContinueOnReturnCodeSet(Set(0))
-    val Memory = 2.0
-    val Preemptible = false
-    val Zones = Vector("us-central1-a")
-  }
-
-  private def validateRuntimeAttributes(attributeMap: AttributeMap): ValidationNel[String, CromwellRuntimeAttributes] = {
-    val docker = attributeMap.get(DOCKER)
-    val zones = attributeMap.get(DEFAULT_ZONES) map { _.split("\\s+").toVector } getOrElse Defaults.Zones
+  val LocalizationDisk = LocalDisk(LocalDiskName, DiskType.SSD, sizeGb=10)
+
+  val defaults: CromwellRuntimeAttributes = {
+    // .get below because we assume that the default values are valid
+    // There's a unit test in CromwellRuntimeAttributeSpec to test these defaults
+    validateRuntimeAttributes(Map(
+      "cpu" -> WdlInteger(1),
+      "defaultDisks" -> WdlString(LocalizationDisk.toString),
+      "defaultZones" -> WdlString("us-central1-a"),
+      "continueOnReturnCode" -> WdlInteger(0),
+      "failOnStderr" -> WdlBoolean.False,
+      "preemptible" -> WdlBoolean.False,
+      "memory" -> WdlString("2GB")
+    )).get
+  }
+
+  private def validateRuntimeAttributes(attributes: Map[String, WdlValue]): Try[CromwellRuntimeAttributes] = {
```

Comment:
ErrorOr. This isn't a public facing function so debate on whether or not a library API should throw or Try doesn't apply here